packages-ts/gauntlet-terra-cw20-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,250 @@
+  currentStatus?: 'pending' | 'open' | 'rejected' | 'passed' | 'executed'
+  data?: WasmMsg[]
+}
+
+export const wrapCommand = (command) => {
+  return class Multisig extends TerraCommand {
+    command: TerraCommand
+    multisig: AccAddress
+    multisigGroup: AccAddress
+
+    static id = `${command.id}:multisig`
+
+    constructor(flags, args) {
+      super(flags, args)
+
+      this.command = new command(flags, args)
+
+      if (!AccAddress.validate(process.env.MULTISIG_ADDRESS)) throw new Error(`Invalid Multisig wallet address`)
+      if (!AccAddress.validate(process.env.MULTISIG_GROUP)) throw new Error(`Invalid Multisig group address`)
+      this.multisig = process.env.MULTISIG_ADDRESS as AccAddress
+      this.multisigGroup = process.env.MULTISIG_GROUP as AccAddress
+    }
+
+    makeRawTransaction = async (signer: AccAddress, state?: State) => {
+      const message = await this.command.makeRawTransaction(this.multisig)
+
+      const operations = {
+        [Action.CREATE]: this.executePropose,
+        [Action.APPROVE]: this.executeApproval,
+        [Action.EXECUTE]: this.executeExecution,
+        [Action.NONE]: () => {
+          throw new Error('No action needed')
+        },
+      }
+
+      if (state.nextAction !== Action.CREATE) {
+        this.require(
+          await this.isSameProposal(state.data, [this.toWasmMsg(message)]),
+          'The transaction generated is different from the proposal provided',
+        )
+      }
+
+      return operations[state.nextAction](signer, Number(this.flags.proposal), message)
+    }
+
+    isSameProposal = (proposalMsgs: WasmMsg[], generatedMsgs: WasmMsg[]) => {
+      return isDeepEqual(proposalMsgs, generatedMsgs)
+    }
+
+    toWasmMsg = (message: MsgExecuteContract): WasmMsg => {
+      return {
+        wasm: {
+          execute: {
+            contract_addr: message.contract,
+            funds: message.coins.toArray().map((c) => c.toData()),
+            msg: Buffer.from(JSON.stringify(message.execute_msg)).toString('base64'),
+          },
+        },
+      }
+    }
+
+    executePropose: ProposalAction = async (signer, _, message) => {
+      logger.info('Generating data for creating new proposal')
+      const proposeInput = {
+        propose: {
+          description: command.id,
+          msgs: [this.toWasmMsg(message)],
+          title: command.id,
+          // TODO: Set expiration time
+          // latest: { never: {} }
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, proposeInput)
+    }
+
+    executeApproval: ProposalAction = async (signer, proposalId) => {
+      logger.info(`Generating data for approving proposal ${proposalId}`)
+      const approvalInput = {
+        vote: {
+          vote: Vote.YES,
+          proposal_id: proposalId,
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, approvalInput)
+    }
+
+    executeExecution: ProposalAction = async (signer, proposalId) => {
```

Comment:
Execute execution? :D

Reply:
None of these are "executing" anything, but just building payloads. Consider renaming.

Reply:
Hehe yes, I'll rename it